L103-core-move-insecure-creds-declaration.md

```diff
@@ -0,0 +1,36 @@
+Move function declarations for insecure credentials from `grpc/grpc_security.h`
+----
+* Author(s): [Cheng-Yu Chung (@ralphchung)](https://github.com/ralphchung)
+* Approver: [@ctiller](https://github.com/ctiller), [@markdroth](https://github.com/markdroth), [@mehrdada](https://github.com/mehrdada), [@drfloob](https://github.com/drfloob)
+* Status: Ready for Implementation
+* Implemented in: C Core
+* Last updated: 09/28/2022
+* Discussion at: https://groups.google.com/g/grpc-io/c/6qvo-UVs-uI
+
+## Abstract
+
+Move function declarations for insecure credentials from `grpc/grpc_security.h` to a new header file `grpc/grpc_insecure_credentials.h`.
```

Comment:
I'm thinking that maybe we should do this not just for insecure creds but also for all credential types.  I think this will be needed anyway to get us to a place where we no longer have any `public_hdr` targets in the BUILD file.  CC @ctiller